VotingApplication/VotingApplication.Web.Api.Tests/E2E/BasicPollTests.cs

```diff
@@ -0,0 +1,603 @@
+                    HiddenResults = false
+                };
+
+                _context.Polls.Add(_defaultBasicPoll);
+                _context.SaveChanges();
+            }
+
+            [ClassCleanup]
+            public static void ClassCleanup()
+            {
+                PollClearer pollTearDown = new PollClearer(_context);
+                pollTearDown.ClearPoll(_defaultBasicPoll);
+                _context.Dispose();
+            }
+
+            [TestMethod, TestCategory("E2E")]
+            public void BasicVote_PopulatedOptions_DisplaysAllOptions()
+            {
+                _driver.Navigate().GoToUrl(SiteBaseUri + "Poll/#/Vote/" + _defaultBasicPoll.UUID);
+                IReadOnlyCollection<IWebElement> optionNames = _driver.FindElements(NgBy.Binding("option.Name"));
+
+                Assert.AreEqual(_defaultBasicPoll.Options.Count, optionNames.Count);
+                CollectionAssert.AreEquivalent(_defaultBasicPoll.Options.Select(o => o.Name).ToList(),
+                                               optionNames.Select(o => o.Text).ToList());
+            }
+
+            [TestMethod, TestCategory("E2E")]
+            public void BasicVote_PopulatedOptions_DisplaysAllOptionDescriptions()
+            {
+                _driver.Navigate().GoToUrl(SiteBaseUri + "Poll/#/Vote/" + _defaultBasicPoll.UUID);
+                IReadOnlyCollection<IWebElement> optionDescriptions = _driver.FindElements(NgBy.Model("option.Description"));
+
+                Assert.AreEqual(_defaultBasicPoll.Options.Count, optionDescriptions.Count);
+                List<String> expectedDescriptions = new List<string>();
+
+                foreach (Option option in _defaultBasicPoll.Options)
+                {
+                    string truncatedDescription = option.Description
+                                                          .Split(' ')
+                                                          .Aggregate((prev, curr) => (curr.Length + prev.Length >= truncatedTextLimit) ?
+                                                                                     prev : prev + " " + curr);
+
+                    if (truncatedDescription != option.Description)
+                    {
+                        truncatedDescription += "... Show More";
+                    }
+                    expectedDescriptions.Add(truncatedDescription);
+                }
+
+                CollectionAssert.AreEquivalent(expectedDescriptions,
+                                               optionDescriptions.Select(o => o.Text).ToList());
+            }
+
+            [TestMethod, TestCategory("E2E")]
+            public void BasicVote_VotingOnOption_NavigatesToResultsPage()
+            {
+                _driver.Navigate().GoToUrl(SiteBaseUri + "Poll/#/Vote/" + _defaultBasicPoll.UUID);
+                IReadOnlyCollection<IWebElement> buttons = _driver.FindElements(By.TagName("Button"));
+                buttons.First(b => b.Text == "Vote").Click();
```

Comment:
Would it not make more sense to give the button an id which we can then reference, instead of having this logic?